### ISSUE TITLE
while `ctrack` command works fine from the command line, `conjuror` has issues finding the right file

### ISSUE THREAD

**Issue by simonv3** [bnvk/Conjuror#40]

**Comment by bnvk** [comment]
Heya! This is definitely something I've thought about a bit and encountered myself. What I'm thinking is the best course of action is storing key/value items in an array in the config file that keep track of "projects" on a users system- so something like

```
"projects": [{
  "Client Work": "/home/user/TimeTracking/datapackage.json",
  "Expenses": "/home/user/Receipts/datapackages.json"
}]
```

The think here is that then from the CLI a user could call `conjuror projects` and get a list of the items to open or run reports on, or conversely `conjuror track` which then the first step is picking from the "projects" the type of data one is tracking...

**Comment by simonv3** [comment]
yeah that makes a lot of sense. I think that's been missing. Then the json is a recipe indicator that could get pretty flexible as well - even to indicate the kind of questions in track?

**Comment by bnvk** [comment]
> Then the json is a recipe indicator

That is precisely how I've always envisioned it and what attracted me to DataPackages- using that JSON schema to help a user pick recipes- automagically saying "this data looks like it belongs on a map, bar chart, invoice, etc..."

**Comment by bnvk** [comment]
I've renamed this issue and am making to-do's of the aspects of the app in which this impacts that need to accounted for when implementing:

- [ ] Allows user to specify a `projects_path` during Setup process
- [ ] The `setup.js` script should recursively autoscan path for existing `.json` files and determine if it is a datapackage to be added to the users config
- [ ] Make the `output.js` script generate its list of projects from this value

@simonv3 just tagging you here so you see this, as I think this is relevant to recipes :wink:

**Comment by simonv3** [comment]
So, something that's been bothering me and I'll tackle once I have some time: I want to be able to properly add a project to my config file, and I can't look at what I've already logged today (because I've been manually creating projects without creating a corresponding .json file). Noting that here because it's the most relevant place.
